control/control_performance_analysis/src/control_performance_analysis_core.cpp

```diff
@@ -210,19 +179,20 @@ bool ControlPerformanceAnalysisCore::calculateErrorVars()
     return false;
   }
 
-  auto && pose_interp_wp_ = pair_pose_interp_wp_.second;
+  const auto pose_interp_wp_ = pair_pose_interp_wp_.second;
 
   // Create interpolated waypoint vector
-  std::vector<double> interp_waypoint_xy{pose_interp_wp_.position.x, pose_interp_wp_.position.y};
+  const std::vector<double> interp_waypoint_xy{
+    pose_interp_wp_.position.x, pose_interp_wp_.position.y};
 
   // Create vehicle position vector
-  std::vector<double> vehicle_position_xy{
+  const std::vector<double> vehicle_position_xy{
     current_vec_pose_ptr_->position.x, current_vec_pose_ptr_->position.y};
 
   // Get Yaw angles of the reference waypoint and the vehicle
-  double target_yaw = tf2::getYaw(pose_interp_wp_.orientation);
-  double vehicle_yaw_angle = tf2::getYaw(current_vec_pose_ptr_->orientation);
-
+  const double target_yaw = tf2::getYaw(pose_interp_wp_.orientation);
```

Comment:
how about `double const & target_yaw` and for others too, instead of copying we can give const refs.